embassy-hal-common/src/lib.rs

```diff
@@ -10,4 +10,5 @@ mod macros;
 mod peripheral;
 pub mod ratio;
 pub mod ring_buffer;
+pub mod stm32;
```

Comment:
`hal-common` is for chip-independent stuff shared between all HALs. It shouldn't have stm32 stuff.

I know there's no way to run the tests there... but still, please put them there, and they'll run if someday we figure out how to run them.

Reply:
I have moved these back - and, I found a way to actually run the tests from within embassy-stm32.